graviti/manager/commit.py

```diff
@@ -251,8 +252,9 @@ def __init__(self, dataset: "Dataset") -> None:
     def _generate(
         self, revision: Optional[str], offset: int, limit: int
```

Comment:
shall we set the type of `revision` to `str` in `_generate` rather than `Optional[str]`

Reply:
move the `revision = head.commit_id` to `list()` instead of `_generate()`? But here the revision could be None, which means the default commit